src/rust/engine/graph/src/node.rs

```diff
@@ -102,4 +103,14 @@ pub trait NodeContext: Clone + Send + 'static {
   /// Returns a reference to the Graph for this Context.
   ///
   fn graph(&self) -> &Graph<Self::Node>;
+
+  ///
+  /// Spawns a Future on an Executor provided by the context.
+  ///
+  /// NB: Unlike the futures `Executor` trait itself, this implementation _must_ spawn the work
+  /// on another thread, as it is called from within the Graph lock.
```

Comment:
Is there a way we can encode this into the type so we can't make this mistake? And/or would it make sense to name this `spawn_separate_thread()`?

Reply:
The bounds (`Send`) sortof already hint that the value will be sent to another thread, but unfortunately that is lost in the noise of _all_ `Futures` currently requiring a `Send` bound. Renaming it would make sense, yea. I'll do that in #6059.